### PR TITLE
feat: add custom user, profile, and Wishlist models

### PR DESCRIPTION
**Pull request by AmrElkfrawy**

### CODE REVIEW

File: api/models/profile.py [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+        preferred_currency: Preferred currency of the user
+    """
+
+    user = models.OneToOneField(User, on_delete=models.CASCADE)
+    USER_TYPES = (
+        ("seller", "Seller"),
+        ("customer", "Customer"),
+    )
+    type = models.CharField(max_length=10, choices=USER_TYPES, default="customer")
+    address = models.CharField(max_length=255, blank=True, null=True)
```
Comment (fadymalak) [inline api/models/profile.py]: it's not  recommend to make both blank and null  equal True

File: api/models/profile.py [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+    phone = models.CharField(max_length=15, blank=True, null=True)
+    preferred_currency = models.CharField(max_length=3, blank=True, null=True)
```
Comment (fadymalak) [inline api/models/profile.py]: add enum choices for currency

File: api/models/user.py [hunk 3]
```diff
@@ -0,0 +1,51 @@
...
+    Fields:
+        email: Email of the user
+        name: Name of the user
+        otp_code: OTP code for user verification
+        otp_created_at: Time when the OTP code was created
+    """
+
+    email = models.EmailField(max_length=255, unique=True)
+    name = models.CharField(max_length=255)
+    otp_code = models.CharField(max_length=6, blank=True, null=True)
```
Comment (fadymalak) [inline api/models/user.py]: it's not recommend to make both blank and null equal True

**Comment by fadymalak** [comment]
**merge from main or master and use Audit model**
try to run makemigration & migrate and test manually, don't push migration files

File: api/models/profile.py (hunk 1 above)
Comment (AmrElkfrawy) [reply api/models/profile.py]: blank equal True is for form validation, e.g. if we want to create something using admin panel, null is database constraint

Comment (fadymalak) [reply api/models/profile.py]: you can make these fields required in form if you keep null=True
or 
make null=False and add default empty value 
@AmrElkfrawy

**Pull request merged**